pyomo/contrib/mindtpy/MindtPy.py

```diff
@@ -239,11 +242,25 @@ def solve(self, model, **kwds):
                     from_list=solve_data.best_solution_found.MindtPy_utils.variable_list,
                     to_list=MindtPy.variable_list,
                     config=config)
-                copy_var_list_values(
-                    MindtPy.variable_list,
-                    [i for i in solve_data.original_model.component_data_objects(
-                        Var) if not i.fixed],
-                    config)
+                # The original does not have variable list. Use get_vars_from_components() should be used for both working_model and original_model to exclude the unused variables.
+                solve_data.working_model.MindtPy_utils.deactivate()
+                if solve_data.working_model.find_component("_int_to_binary_reform") is not None:
+                    solve_data.working_model._int_to_binary_reform.deactivate()
+                copy_var_list_values(list(get_vars_from_components(block=solve_data.working_model, 
+                                         ctype=(Constraint, Objective), 
+                                         include_fixed=False, 
+                                         active=True,
+                                         sort=True, 
```

Comment:
Here, I need to set `sort=Ture`. Otherwise, the order of variables in `list(get_vars_from_components(block=solve_data.working_model, ...)` and `list(get_vars_from_components(block=solve_data.original_model, ...)` will be different.
@emma58